### PR TITLE
new design for course card

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

part of #1997

#### What's this PR do?

This PR implements a new `CourseCard` component for displaying courses and uses it on the course home page (`/courses`). It should match what's showed in the updated designs (see on invision) with the exception of the 'favorite' button which isn't implemented yet.

#### How should this be manually tested?

check out the course home page and make sure things look good. we'll change over the course search result page to use this card for display search results in a separate PR.

#### Screenshots (if appropriate)

featuredwow

### CODE REVIEW

File: static/js/components/CourseCard.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          ))
+          : null}
+      </div>
+      <div className="row availability">
+        {availabilityLabel(course.availability)}
+      </div>
+      <div className="row platform">
+        <img
+          className="course-platform"
+          src={platformLogoUrls[course.platform]}
```
Comment (rhysyngsun) [inline static/js/components/CourseCard.js]: Could we have an `alt` attribute here?

File: static/js/components/CourseCard.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        />
+      </div>
+      <div className="row price">{minPrice(course)}</div>
+    </div>
+    <div className="blue-thing" />
```
Comment (rhysyngsun) [inline static/js/components/CourseCard.js]: Nit: something more descriptive like `blue-bottom-border`?

Comment (aliceriot) [reply static/js/components/CourseCard.js]: haha, I suppose that is a little more descriptive than `.blue-thing`...

File: static/js/components/CourseCard.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/CourseCard.js]: I added one here and then went ahead and added one to the 'cover image' above as well

**Comment by rhysyngsun** [comment]
A few things on functionality:

- The heights of the different cards aren't equal under firefox
- The vertical padding around the platform logo needs to be increased

Screenshot_2019-07-12 MIT Open Learning

**Comment by rhysyngsun** [comment]
Pulled the last commit and it's a little bit different but same general problems:

Screenshot_2019-07-12 MIT Open Learning(1)

**Comment by aliceriot** [comment]
yeah the problem is the courses with three longer subject names, I was testing only with shorter ones I think. I'll fix it up

**Comment by rhysyngsun** [comment]
Oh I was thinking it was the number of topics

**Comment by aliceriot** [comment]
ok that should be addressed, I fixed the display for the long three-category courses and also fixed the font styling.

File: static/js/lib/constants.js [hunk 3]
```diff
@@ -11,9 +11,15 @@ export const COURSE_AVAILABLE_NOW = "Available Now"
 export const COURSE_ARCHIVED = "Archived"
 export const COURSE_PRIOR = "Prior"
 
+const ocwPlatform = "ocw"
+const edxPlatform = "mitx"
 export const platforms = {
-  OCW: "ocw",
-  edX: "mitx"
+  OCW: ocwPlatform,
+  edX: edxPlatform
```
Comment (rhysyngsun) [inline static/js/lib/constants.js]: Missed this on my first pass, but this needs a set of constants and a platform image for bootcamps as well.

Comment (aliceriot) [reply static/js/lib/constants.js]: right I meant to ask about that but it slipped my mind - what should the value of the constant be?

Comment (rhysyngsun) [reply static/js/lib/constants.js]: `"bootcamps"`

**Comment by aliceriot** [comment]
@rhysyngsun ok I just added that and then added in the micromasters one as well

**Pull request merged**